app/presenters/two_factor_auth_code/generic_delivery_mode_presenter.rb

```diff
@@ -0,0 +1,42 @@
+module TwoFactorAuthCode
+  class GenericDeliveryModePresenter
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TranslationHelper
+    include Rails.application.routes.url_helpers
+
+    def initialize(data_model)
+      data_model.each do |k, v|
+        instance_variable_set("@#{k}", v)
+        self.class.send(:attr_reader, k.to_sym)
+      end
+    end
+
+    def header
+      raise NotImplementedError
+    end
+
+    def help_text
+      raise NotImplementedError
+    end
+
+    def authentication_fallback
+      raise NotImplementedError
+    end
+
+    def fallback_options_links
+      raise NotImplementedError
+    end
+
+    def recovery_code_link
+      t('devise.two_factor_authentication.recovery_code_fallback.text', link: recovery_code_tag)
+    end
+
+    private
+
+    def recovery_code_tag
+      content_tag(:a,
```

Comment:
Requiring that helper produces this error:
`undefined local variable or method `controller' for #`

Trying to include ActionController::Base.helper.link_to produces this one:
`arguments passed to url_for can't be handled`

This is done using the `include` method, is there some other pattern that will mix in these helpers in such a way that an error doesn't occur?

For now, I've stuck a link_to method on the generic presenter that proxies to content_tag.